src/clib/pioc_support.c

```diff
@@ -876,6 +877,453 @@ int PIOc_readmap_from_f90(const char *file, int *ndims, int **gdims, PIO_Offset
+                             int num_tasks, int *task_maplen, int *map, const char *title,
+                             const char *history, int fortran_order)
+{
+    iosystem_desc_t *ios;
+    int max_maplen = 0;
+    int ncid;
+    int ret;
+
+    /* Get the IO system info. */
+    if (!(ios = pio_get_iosystem_from_id(iosysid)))
+        return pio_err(NULL, NULL, PIO_EBADID, __FILE__, __LINE__);
+
+    /* Check inputs. */
+    if (!filename || !global_dimlen || !task_maplen)
+        return pio_err(ios, NULL, PIO_EINVAL, __FILE__, __LINE__);
+    if (title)
+        if (strlen(title) > PIO_MAX_NAME)
+            return pio_err(ios, NULL, PIO_EINVAL, __FILE__, __LINE__);
+    if (history)
+        if (strlen(history) > PIO_MAX_NAME)
+            return pio_err(ios, NULL, PIO_EINVAL, __FILE__, __LINE__);
+
+    LOG((2, "pioc_write_nc_decomp_int iosysid = %d filename = %s ndims = %d num_tasks = %d",
+         iosysid, filename, ndims, num_tasks));
+
+    /* Find the maximum maplen. */
+    for (int t = 0; t < num_tasks; t++)
+        if (task_maplen[t] > max_maplen)
+            max_maplen = task_maplen[t];
+    LOG((3, "max_maplen = %d", max_maplen));
+
+    /* Create the netCDF decomp file. */
+    if ((ret = PIOc_create(iosysid, filename, NC_WRITE, &ncid)))
+        return pio_err(ios, NULL, ret, __FILE__, __LINE__);
+
+    /* Write an attribute with the version of this file. */
+    char version[] = VERSNO2;
+    if ((ret = PIOc_put_att_text(ncid, NC_GLOBAL, DECOMP_VERSION_ATT_NAME,
+                                 strlen(version) + 1, version)))
+        return pio_err(ios, NULL, ret, __FILE__, __LINE__);
+
+    /* Write an attribute with the max map len. */
+    if ((ret = PIOc_put_att_int(ncid, NC_GLOBAL, DECOMP_MAX_MAPLEN_ATT_NAME,
+                                PIO_INT, 1, &max_maplen)))
+        return pio_err(ios, NULL, ret, __FILE__, __LINE__);
+
+    /* Write title attribute, if the user provided one. */
+    if (title)
+        if ((ret = PIOc_put_att_text(ncid, NC_GLOBAL, DECOMP_TITLE_ATT_NAME,
+                                     strlen(title) + 1, title)))
+            return pio_err(ios, NULL, ret, __FILE__, __LINE__);
+
+    /* Write history attribute, if the user provided one. */
+    if (history)
+        if ((ret = PIOc_put_att_text(ncid, NC_GLOBAL, DECOMP_HISTORY_ATT_NAME,
+                                     strlen(history) + 1, history)))
+            return pio_err(ios, NULL, ret, __FILE__, __LINE__);
+
+    /* Write a source attribute. */
+    char source[] = "Decomposition file produced by PIO library.";
+    if ((ret = PIOc_put_att_text(ncid, NC_GLOBAL, DECOMP_SOURCE_ATT_NAME,
+                                 strlen(source) + 1, source)))
+        return pio_err(ios, NULL, ret, __FILE__, __LINE__);
+
+    /* Write an attribute with array ordering (C or Fortran). */
+    char c_order_str[] = DECOMP_C_ORDER_STR;
+    char fortran_order_str[] = DECOMP_FORTRAN_ORDER_STR;
```

Comment:
There is a distinction because if you write it out in C and read it in Fortran things may get confused.

(I think. Am I wrong about that?)